src/Orleans/Facet/FacetAttribute.cs

```diff
@@ -0,0 +1,12 @@
+﻿
+using System;
+
+namespace Orleans
+{
+    /// <summary>
+    /// Marker attribute for facets
+    /// </summary>
+    public abstract class FacetAttribute : Attribute
```

Comment:
We don't necessarily need a base attribute, but I'm not convinced that we shouldn't have some kind of static marker (attribute on attribute, interface on attribute, base class on attribute) - just to make things obvious. Attribute on attribute would require more work on this PR since it cannot be used as a generic type constraint and generally requires more dynamic checking.

If we went with an interface instead, we could more easily change the requirement later to open it up to any attribute type later. That way we're narrow on what we accept today but we have the option to change our decision later without breaking anyone.

How does that sound?